mingw-w64-curl/0001-Make-cURL-relocatable.patch

```diff
@@ -156,7 +156,7 @@ index 000000000..c09458e95
 +    *path_p = '/';
 +  }
 +  /* Replace any '//' with '/' */
-+  path_p = path;
++  path_p = path + !!*path; /* skip first character, if any, to handle UNC paths correctly */
 +  while ((path_p = strstr (path_p, "//")) != NULL)
 +  {
 +    memmove (path_p, path_p + 1, path_size--);
```

Comment:
I know it's unrelated to the change, but it looks like a bug to me.
This memmove() call will modify data located after the "path" string in memory for most of the cases.
Like if the "path" is "foobar//goo" it will be called like memmove("//goo", "/goo", 11), so it will move data located after the end of the "path" string unless the memmove() has a specific implementation to check end-of-string (that would be quite unusual).

I'd expect it to be something like:

memmove (path_p, path_p + 1, strlen(path_p + 1) + 1);
or to be shorter:
memmove (path_p, path_p + 1, strlen(path_p));

Reply:
I guess you're right... This bug exists in MSYS2, too.

Reply:
Probably a much better version would look like this:

```c
#define IS_SLASH(c) ((c) == '/' || (c) == '\\')

void
sanitize_path(char *path)
{
  int i, j = IS_SLASH(*path) ? 1 : 0;

  for (i = j; path[i]; i++) {
    if (!IS_SLASH(path[i]))
      path[j++] = path[i];
    else {
      /* reduce runs of slashes into a single forward slash */
      while (IS_SLASH(path[i+1]))
        i++;
      path[j++] = '/';
    }
  }

  path[j] = '\0';
}
```

Reply:
Agree

Reply:
Let's try to set aside time at some point in the near future to get this one in.